src/app/api/v1/fxa-rp-events/route.ts

```diff
@@ -293,12 +321,35 @@ export async function POST(request: NextRequest) {
               MONITOR_PREMIUM_CAPABILITY,
             )
           ) {
+            // Update fxa profile data to match subscription status.
+            // This is done before trying to deactivate the OneRep subscription, in case there are
+            // any problems with deactivation.
+            await changeSubscription(subscriber, false);
+
+            // MNTOR-2103: if one rep profile id doesn't exist in the db, fail immediately
+            if (!oneRepProfileId) {
+              logger.error("onerep_profile_not_found", {
+                subscriber_id: subscriber.id,
+              });
+
+              captureException(
+                new Error(`No OneRep profile Id found, subscriber: ${
+                  subscriber.id as string
+                }\n
+                        Event: ${event}\n
+                        updateFromEvent: ${JSON.stringify(
+                          updatedSubscriptionFromEvent,
+                        )}`),
+              );
+              return NextResponse.json(
+                { success: false, message: "failed_activating_subscription" },
+                { status: 500 },
+              );
+            }
+
             // deactivation stops opt out process
             await deactivateProfile(oneRepProfileId);
```

Comment:
Should we do the same with deactivate to catch for the 403?

Reply:
There isn't anything else to do after deactivation (there's no inverse of the "opt-out" API call), deactivate should also stop the opt-out process.

Reply:
Oh I see what you're saying though, if it's already de-activated. Yes I think it does make sense then.